applications/FluidDynamicsApplication/custom_elements/fluid_adjoint_element.h

```diff
@@ -0,0 +1,304 @@
+        VectorType& rValues,
+        int Step) const override;
+
+    void Initialize(const ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateLocalSystem(
+        MatrixType& rLeftHandSideMatrix,
+        VectorType& rRightHandSideVector,
+        const ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateLeftHandSide(
+        MatrixType& rLeftHandSideMatrix,
+        const ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateRightHandSide(
+        VectorType& rRightHandSideVector,
+        const ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateLocalVelocityContribution(
+        MatrixType &rDampMatrix,
+        VectorType &rRightHandSideVector,
+        const ProcessInfo &rCurrentProcessInfo) override;
+
+    void CalculateFirstDerivativesLHS(
+        MatrixType& rLeftHandSideMatrix,
+        const ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateSecondDerivativesLHS(
+        MatrixType& rLeftHandSideMatrix,
+        const ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateMassMatrix(
+        MatrixType& rMassMatrix,
+        const ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateDampingMatrix(
+        MatrixType& rDampingMatrix,
+        const ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateSensitivityMatrix(
+        const Variable<array_1d<double, 3>>& rSensitivityVariable,
+        Matrix& rOutput,
+        const ProcessInfo& rCurrentProcessInfo) override;
+
+    void Calculate(
+        const Variable<Vector>& rVariable,
+        Vector& rOutput,
+        const ProcessInfo& rCurrentProcessInfo) override;
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override;
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const override;
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override;
+
+    ///@}
+
+protected:
+    ///@name Protected Members
+    ///@{
+
+    ConstitutiveLaw::Pointer mpConstitutiveLaw = nullptr;
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    void AddFluidResidualsContributions(
+        VectorType& rResidual,
+        const ProcessInfo& rCurrentProcessInfo);
+
+    void AddFluidFirstDerivatives(
+        MatrixType& rDerivativesMatrix,
+        const ProcessInfo& rCurrentProcessInfo,
+        const double MassTermsDerivativesWeight = 1.0);
+
+    void AddFluidSecondDerivatives(
+        MatrixType& rDerivativesMatrix,
+        const ProcessInfo& rCurrentProcessInfo);
+
+    void AddFluidShapeDerivatives(
+        MatrixType& rDerivativesMatrix,
+        const ProcessInfo& rCurrentProcessInfo);
+
+    void CalculateGeometryData(
+        Vector& rGaussWeights,
+        Matrix& rNContainer,
+        ShapeFunctionDerivativesArrayType& rDN_DX,
+        const GeometryData::IntegrationMethod& rIntegrationMethod) const;
```

Comment:
Doxygen documentation for these methods is missing.

Reply:
done :)